redis/tests/support/mod.rs

```diff
@@ -336,6 +339,40 @@ where
         }
         Value::Okay => write!(writer, "+OK\r\n"),
         Value::Status(ref s) => write!(writer, "+{s}\r\n"),
+        Value::Map(ref values) => {
+            write!(writer, "%{}\r\n", values.len())?;
```

Comment:
notice that Bulk, Map, Set, and Push all use the same logic, with the difference of a single char in the beginning. Maybe try reducing the repetition?

Reply:
Now they all use same function except Push which requires special care now...